src/linear_algebra/exterior_algebra/of_alternating.lean

```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2022 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+
+import linear_algebra.clifford_algebra.fold
+import linear_algebra.exterior_algebra.basic
+
+/-!
+# Extending an alternating map to the exterior algebra
+
+## Main definitions
+
+* `exterior_algebra.lift_alternating`: construct a linear map out of the exterior algebra
+  given alternating maps (corresponding to maps out of the exterior powers).
+* `exterior_algebra.lift_alternating_equiv`: the above as a linear equivalence
+
+## Main results
+
+* `exterior_algebra.lhom_ext`: linear maps from the exterior algebra agree if they agree on the
+  exterior powers.
+
+-/
+
+variables {R M N N' : Type*}
+variables [comm_ring R] [add_comm_group M] [add_comm_group N] [add_comm_group N']
+variables [module R M] [module R N] [module R N']
+
+-- This instance can't be found where it's needed if we don't remind lean that it exists.
+instance alternating_map.module_add_comm_group {ι : Type*} [decidable_eq ι] :
+  module R (alternating_map R M N ι) :=
+by apply_instance
+
+namespace exterior_algebra
+open clifford_algebra (hiding ι)
+
+/-- Build a map out of the exterior algebra given a collection of alternating maps acting on each
+exterior power -/
+def lift_alternating : (Π i, alternating_map R M N (fin i)) →ₗ[R] exterior_algebra R M →ₗ[R] N :=
```

Comment:
It is important that this map respects the grading. Should this fact be established later or should we define:
```lean
local notation `Λ` i := (pow (exterior_algebra.ι R : M →ₗ[R] exterior_algebra R M).range i : submodule R _)

def lift_alternating : alternating_map R M N (fin i) →ₗ[R] (Λ i) →ₗ[R] N :=
```
here instead?

Reply:
Your version looks like a weaker version of my version, since you can obtain it by composing my version with `linear_map.proj i` and `submodule.subtype (Λ i)` on the left and right arguments respectively.

What exactly do you have in mind for showing it "respects" the gradings? Is `lift_alternating_apply_ι_multi` strong enough, or did you have a stronger statement in mind?

Reply:
> Your version looks like a weaker version of my version, since you can obtain it by composing my version with `linear_map.proj i` and `submodule.subtype (Λ i)` on the left and right arguments respectively.

Possibly I'm being dense, but I don't actually see how to do this. However, it's not so important.

> What exactly do you have in mind for showing it "respects" the gradings? Is `lift_alternating_apply_ι_multi` strong enough, or did you have a stronger statement in mind?

I mean that we should have the following equivalence:
```lean
alternating_map R M N (fin i) ≃ₗ[R] (Λ i) →ₗ[R] N
```
If we can obtain this from what you have then I'm happy (modulo possible future work to bundle this as a not-yet-defined `graded_algebra_equiv` type).

Reply:
Ah sorry, I meant compose with `linear_map.single`. I'm a little unsure what you're thinking of for `graded_algebra_equiv`, but I agree it would be future work.